Classes/UserFunctions/GridRecords.php

```diff
@@ -20,24 +22,26 @@ final class GridRecords
      *
      * @param string $content Empty string (no content to process)
      * @param array $conf TypoScript configuration
+     * @param ServerRequestInterface $request
      * @return string Comma-separated list of colPos values associated to a grid column on the current page
-     * @throws \Doctrine\DBAL\DBALException
-     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws Exception
      */
-    public function getColPosList(string $content, array $conf): string
+    public function getColPosList(string $content, array $conf, ServerRequestInterface $request): string
     {
         $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
+        $currentContentObject = $request->getAttribute('currentContentObject');
+        $pid = $currentContentObject->data['uid'];
         $result = $queryBuilder
-            ->select('colPos', 'tx_container_parent')
+            ->select('colPos','pid', 'tx_container_parent')
             ->from('tt_content')
             ->where(
                 $queryBuilder->expr()->eq(
                     'pid',
-                    $queryBuilder->createNamedParameter($GLOBALS['TSFE']->contentPid, \PDO::PARAM_INT)
+                    $pid
```

Comment:
I know that `createNamedParameter` is only mandatory when dealing with user input but we tend to use this as a default.

```suggestion
                    $queryBuilder->createNamedParameter($pid, Connection::PARAM_INT)
```